ipatests/test_integration/test_service_permissions.py

```diff
@@ -78,61 +78,68 @@ def test_service_as_user_admin(self):
 
-        service_name = 'testservice/%s@%s' % (self.master.hostname,
+        service_name2 = 'testservice2/%s@%s' % (self.master.hostname,
                                               self.master.domain.realm)
 
-        keytab_file = os.path.join(self.master.config.test_dir,
-                                   'testservice_keytab')
+        keytab_file2 = os.path.join(self.master.config.test_dir,
+                                   'testservice_keytab2')
 
         # Prepare a service without authentication indicator
-        self.master.run_command(['ipa', 'service-add', service_name])
+        self.master.run_command(['ipa', 'service-add', service_name2])
 
         self.master.run_command(['ipa-getkeytab',
-                                 '-p', service_name,
-                                 '-k', keytab_file])
+                                 '-p', service_name2,
+                                 '-k', keytab_file2])
 
         # Set authentication-type for admin user
         self.master.run_command(['ipa', 'user-mod', 'admin',
                                  '--user-auth-type=password',
                                  '--user-auth-type=otp'])
 
         # Authenticate
-        self.master.run_command(['kinit', '-k', service_name,
-                                 '-t', keytab_file])
+        self.master.run_command(['kinit', '-k', service_name2,
+                                 '-t', keytab_file2])
 
         # Verify access to service is granted
-        result = self.master.run_command(['kvno', service_name],
+        result = self.master.run_command(['kvno', service_name2],
                                          raiseonerr=False)
         assert result.returncode == 0
 
         # Obtain admin ticket to be able to update service
         tasks.kinit_admin(self.master)
 
         # Modify service to have authentication indicator
-        self.master.run_command(['ipa', 'service-mod', service_name,
+        self.master.run_command(['ipa', 'service-mod', service_name2,
                                  '--auth-ind=otp'])
 
         self.master.run_command(['ipa-getkeytab',
-                                 '-p', service_name,
-                                 '-k', keytab_file])
+                                 '-p', service_name2,
+                                 '-k', keytab_file2])
 
         # Authenticate
-        self.master.run_command(['kinit', '-k', service_name,
-                                 '-t', keytab_file])
+        self.master.run_command(['kinit', '-k', service_name2,
+                                 '-t', keytab_file2])
 
         # Verify access to service is rejected
-        result = self.master.run_command(['kvno', service_name],
+        result = self.master.run_command(['kvno', service_name2],
                                          raiseonerr=False)
         assert result.returncode > 0
+
+    def test_service_del(self):
+        """ Test that host can remove there own services.
```

Comment:
nit: replace "there" with "their" or "it's"

Not sure if you need/want to mention it here but this is also exercising that hosts can add their own services as well.